Add alias generators to pydantic

dmontagu opened the pull request:
@Kludex what do I need to update with regard to import redirects etc.?

Also:
* Do I need to set `__all__` in the new module?
* Should I import these alias generators directly into `pydantic`, or leave as-is?
* Should I rename the functions? Might be good to rename to `to_pascal` and `to_camel` instead of `to_camel` and `to_lower_camel`?

Closes #5560 
Closes #5241 

Selected Reviewer: @samuelcolvin

dmontagu commented:
please review

samuelcolvin commented on pydantic/alias_generators.py:
```diff
@@ -0,0 +1,26 @@
+import re
```
please add these to the API documentation.

samuelcolvin reviewed (commented):
otherwise LGTM.

* Do I need to set `__all__` in the new module?

yes please

* Should I import these alias generators directly into `pydantic`, or leave as-is?

humm, I'm not in favour of adding everything to `pydantic`, but I do see it's confusing that a few things are not included, so I guess add them.

* Should I rename the functions? Might be good to rename to `to_pascal` and `to_camel` instead of `to_camel` and `to_lower_camel`?

yes please.

samuelcolvin commented:
please update.

dmontagu commented:
@samuelcolvin I believe I have addressed all your feedback with the exception that, given the recent discussions about including _everything_ in `pydantic` directly, I didn't (yet) add the alias generators directly into pydantic. Considering these functions were in `pydantic.utils` in v1, and were not available in `pydantic` directly, I think it makes sense to keep them in a special module.

However, this is still only a weak feeling that it's better to leave them not imported in `pydantic.__init__`; if you think it's best to just add it there (and to the `pydantic.__all__`), I have no problem doing that.

dmontagu commented:
please reviewt
